src/content/en/updates/2017/10/animated-blur.md

```diff
@@ -0,0 +1,187 @@
+  type="video/webm; codecs=vp8"> <source
+  src="https://storage.googleapis.com/webfundamentals-assets/animated-blur/demo_x264.mp4"
+  type="video/mp4; codecs=h264"> </video>
+
+However, this technique can be quite jarring when applied without any
+transitional period. Animating a blur — transitioning from unblurred to blurred
+— seems like a reasonable choice, but if you've ever tried doing this on the
+web, you probably found that the animations are anything but smooth, as this
+[demo](https://googlechromelabs.github.io/ui-element-samples/animated-blur/reallybadblur.html)
+shows if you don't have a powerful machine. Can we do better?
+
+## The problem
+
+<img src="/web/updates/images/2017/10/animated-blur/pipeline.png" alt="Markup is
+  turned into textures by the CPU. Textures are uploaded to the GPU. The GPU
+  draws these textures to the framebuffer using shaders. The blurring happens in
+  the shader." class="attempt-right">
+
+As of now, we can't make animating a blur work efficiently. We can, however,
+find a work-around that looks *good enough*, but is, technically speaking, not
+an animated blur. To get started, let's first understand *why* the animated blur
+is slow. To blur elements on the web there're two techniques: The CSS `filter`
+property and SVG filters. Thanks to increased support and ease of use, CSS
+filters are typically used. However, if you are required to support Internet
+Explorer, you have no choice but to use SVG filters as IE 10 and 11 support
+those but not CSS filters. The good news is that our workaround for animating a
+blur works with both techniques. So let's try to find the bottleneck by looking
+at DevTools:
+
+If you enable "Paint Flashing" in DevTools, it does not flash at all. It looks
+like no repaints are happening. And it's technically correct as a "repaint"
+refers to the CPU having to repaint the texture of an promoted element. Whenever
+an element is both promoted *and* blurred, the blur is applied by the GPU using
+a shader.
+
+Both SVG filters and CSS filters use [convolution
+filters](https://en.wikipedia.org/wiki/Kernel_(image_processing)) to apply a
+blur. Convolution filters are fairly expensive as for every output pixel a
+number of input pixels have to be considered. The bigger the image or the bigger
+the blur radius, the more costly the effect is.
+
+And that's where the problem lies, we are running a rather expensive GPU
+operation every frame, blowing our frame budget of 16ms and therefore ending up
+well below 60fps.
+
+
+## Down the rabbit hole
+
+So what can we do to make this run smoothly? We can use sleight of hand! Instead
+of animating the actual blur value (the radius of the blur), we pre-compute a
+couple of blurred copies where the blur value increases exponentially and
+cross-fade between them using `opacity`.
+
+The cross-fade is a series of overlapping opacity fade-ins and fade-outs. If we
```

Comment:
The English word I'd use to describe the actions in this paragraph is 'cascading'. No action required here. I just thought you might like the tip.